zkevm-circuits/src/evm_circuit/execution/push.rs

```diff
@@ -242,9 +238,10 @@ mod test {
 
     #[test]
     fn push_gadget_simple() {
+        #[cfg(feature = "shanghai")]
```

Comment:
It looks like shanghai is already baked in the circuit logic and hard to separate with a flag now.
```suggestion
```